hooks/useYouTubeChannel.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+import { useArtistProvider } from "@/providers/ArtistProvider";
+import { useUserProvider } from "@/providers/UserProvder";
+
+const useYouTubeChannel = () => {
+  const { selectedArtist } = useArtistProvider();
+  const { userData } = useUserProvider();
+  const [channelData, setChannelData] = useState<string | null>(null);
+
+  useEffect(() => {
+    const artistAccountId = selectedArtist?.account_id;
+    const accountId = userData?.id;
+
+    if (!artistAccountId || !accountId) {
+      setChannelData(null);
+      return;
+    }
+
+    const fetchChannelInfo = async () => {
+      const params = new URLSearchParams({
+        artistAccountId,
+        accountId,
+      });
+      const res = await fetch(`/api/youtube/channel-info?${params.toString()}`);
```

Comment:
single responsibility principle
- actual: fetch is defined in the hook
- required: new lib for the fetch lib/youtube/fetchYouTubeChannelInfo.ts